table/block_based/block.cc

```diff
@@ -717,7 +716,8 @@ int IndexBlockIter::CompareBlockKey(uint32_t block_index, const Slice& target) {
     return 1;  // Return target is smaller
   }
   Slice block_key(key_ptr, non_shared);
-  return Compare(block_key, target);
+  raw_key_.SetInternalKey(block_key, false /* copy */);
+  return comparator_->Compare(applied_key_.UpdateAndGetKey(), target);
```

Comment:
We should change the name from CompareBlockKey if it's now (intentionally) comparing with internal key.

Reply:
Hm, I am confused, this function feels like it needs to be able to handle either internal key or user key (index key could be either one depending on the `format_version`). Calling `SetInternalKey()` above looks like a mistake.

Reply:
Done (call `SetKey()` instead).